src/Type/Accessory/AccessoryArrayListType.php

```diff
@@ -160,6 +167,13 @@ public function setExistingOffsetValueType(Type $offsetType, Type $valueType): T
 			return $this;
 		}
 
+		if (
+			$valueType->isArray()->yes()
+			&& IntegerRangeType::fromInterval(0, null)->isSuperTypeOf($offsetType)->yes()
+		) {
+			return $this;
+		}
+
 		return new ErrorType();
```

Comment:
I think this method can just return `$this` in any case. Assigning existing offset on list will preserve list.